framework/include/cppmicroservices/ServiceFactory.h

```diff
@@ -70,25 +65,35 @@ class ServiceFactory
   virtual ~ServiceFactory() {}
 
   /**
-   * Creates a new service object.
+   * Returns a service object for a bundle.
    *
-   * <p>
-   * The Framework invokes this method the first time the specified
-   * <code>bundle</code> requests a service object using the
+   * The framework invokes this method the first time the specified
+   * \c bundle requests a service object using the
    * <code>BundleContext::GetService(const ServiceReferenceBase&)</code> method. The
-   * service factory can then return a specific service object for each
+   * factory can then return a customized service object for each bundle.
+   *
+   * The framework checks that the returned service object is valid. If the
+   * returned service object is null or does not contain entries for all the
+   * classes named when the service was registered, a framework event of type
+   * \c FrameworkEvent::FRAMEWORK_ERROR is fired containing a service exception
+   * of type \c ServiceException::FACTORY_ERROR and null is returned to the
+   * bundle. If this method throws an exception, a framework event of type
+   * \c FrameworkEvent::FRAMEWORK_ERROR is fired containing a service exception
+   * of type \c ServiceException::FACTORY_EXCEPTION with the thrown exception
+   * as a nested exception and null is returned to the bundle. If this method is
+   * recursively called for the specified bundle, a framework event of type
+   * \c FrameworkEvent::FRAMEWORK_ERROR is fired containing a service exception
+   * of type \c ServiceException::FACTORY_RECURSION and null is returned to the
    * bundle.
    *
-   * <p>
-   * The framework caches the value returned (unless the InterfaceMap is empty),
-   * and will return the same service object on any future call to
-   * <code>BundleContext::GetService</code> for the same bundles. This means the
-   * framework does not allow this method to be concurrently called for the
-   * same bundle.
+   * The framework caches the valid service object, and will return the same
+   * service object on any future call to \c BundleContext::GetService for the
+   * specified bundle. This means the framework does not allow this method to
```

Comment:
The wording is basically taken from the specs (the API documentation). "Whom"  are the clients getting service objects. They indirectly call this function via the framework API.